### ISSUE TITLE
Fields API misses `token_count` fields inside `nested` fields

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#68983]
This is an issue that came up during work on #67432 and was decided to treat as a follow-up issue.
As of 7.11 (that is before #67432) we are missing fields mapped as `token_count` if they are inside `nested` fields.

e.g.
```
DELETE my-index

PUT my-index
{
  "mappings": {
    "properties": {
      "obj": {
        "type": "nested",
        "properties": {
          "name": {
            "type": "text",
            "fields": {
              "length": {
                "type": "token_count",
                "analyzer": "standard"
              }
            }
          }
        }
      }
    }
  }
}

PUT /my-index/_doc/1
{
  "obj": {"name" : "Long John Silver"}
}

POST /my-index/_search
{
  "_source": false,
  "fields" : ["*"]
}
```
 is missing the `obj.name.length` field which is there if e.g. `obj` is a regular `object` field. This is probably due to the fact that the value fetcher used by that field type is a DocValueFetcher and not the usual SourceValueFetcher.

**Comment by cbuescher** [comment]
After talking to @romseygeek, at least at the moment the lookup of docvalues from nested docs might be possible with the changes he is looking into in the course of #68984, so tagging this as a dependency. In the meantime, currently we are getting NPEs when a "token_count" is defined inside a nested field due not passing on the LeadReaderContext to the value fetchers if they are inside a NestedValueFetcher. I'm going to open a PR to adress this for the short-term by returning an empty list and look into adressing the underlying issue mentioned here after #68984 has progressed.

**Comment by jtibshirani** [comment]
@cbuescher it's good that we followed up after #67432 to fix the NPE, but I'm not sure it's worth investing in properly fixing right now. It's currently a very rare case that we can load from doc values within a nested document. Maybe we could simply document this shortcoming, and revisit it if and when it becomes more common to load from doc values.
